src/Controller/ProductReviewController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\OrderRepository;
+use App\Repository\ProductRepository;
+use App\Form\ProductFeedbackType;
+use App\Entity\ProductFeedback;
+use Doctrine\ORM\EntityManagerInterface;
+/**
+ * @Route("product/review")
+ */
+class ProductReviewController extends AbstractController
+{
+    /**
+     * @Route("/{token}", name="product_review", methods={"GET","POST"}, host="buyer.%domain%")
+     */
+    public function index(Request $request, string $token, EntityManagerInterface $entityManager, OrderRepository $orderRepository, ProductRepository $productRepository): Response
+    {
+        $productFeedback = new ProductFeedback();
+        $form = $this->createForm(ProductFeedbackType::class, $productFeedback);
+        $form->handleRequest($request);
+
+        $errors = [];
+        $orderProduct = null;
+        $order = $orderRepository->findOneBy([
+            'token' => $token
+        ]);
+
+        if (!$order) {
+            $errors['invalid_order_token'] = true;
+        } else {
+            $reviewedProducts = $order->getReviewedProductIds();
+            $orderProducts = $order->getOrderProductsIds();
+            $unReviewedProducts = array_diff($orderProducts, $reviewedProducts); 
+            $orderProduct = (!empty($unReviewedProducts)) ? $productRepository->find(current($unReviewedProducts)) : null;
```

Comment:
var names are confusing.
`$reviewedProducts`, `$orderProducts`, `$unReviewedProducts` hold the product IDs. The names should depict this. `reviewedProductIds`, `$orderProductIds` etc are better names.

The name `orderProduct` gives the idea that it would be a single object but in fact this object can hold a list of product objects. Therefore the correct name would be plural `unreviewedProducts`.

Additionally, please refactor this part of code into a separate function eg. `get_unreviewed_products`

Reply:
$orderProduct = (!empty($unReviewedProductIds)) ? $productRepository->find(current($unReviewedProductIds)) : null;

It contains just one object as current() is being used. I'm changing its name to $unreviewedProduct